lxd/instance/drivers/driver_qemu.go

```diff
@@ -1313,6 +1313,136 @@ func (d *qemu) start(stateful bool, op *operationlock.InstanceOperation) error {
+			}
+
+			if slices.Contains(features, "sme") && slices.Contains(features, "sev") {
+				// Get the SEV guest `cbitpos` and `reducedPhysBits`.
+				var cbitpos, reducedPhysBits int
+				var cbitposOK, reducedPhysBitsOK bool
+				for _, feature := range features {
+					if cbitposOK && reducedPhysBitsOK {
+						break
+					}
+
+					if strings.HasPrefix(feature, "sev.cbitpos=") {
+						cbitpos, err = strconv.Atoi(strings.TrimPrefix(feature, "sev.cbitpos="))
+						if err != nil {
+							op.Done(err)
+							return err
+						}
+
+						cbitposOK = true
+					}
+
+					if strings.HasPrefix(feature, "sev.reduced_phys_bits=") {
+						reducedPhysBits, err = strconv.Atoi(strings.TrimPrefix(feature, "sev.reduced_phys_bits="))
+						if err != nil {
+							op.Done(err)
+							return err
+						}
+
+						reducedPhysBitsOK = true
+					}
+				}
+
+				// Write user's dh-cert and session-data to file descriptors.
+				var dhCertFD, sessionDataFD int
+				var dhCertFDOk, sessionDataFDOk bool
+				if d.localConfig["security.sev.session.dh"] != "" {
+					dhCert, err := os.CreateTemp("", "lxd_sev_dh_cert_")
+					if err != nil {
+						op.Done(err)
+						return err
+					}
+
+					err = os.Remove(dhCert.Name())
+					if err != nil {
+						op.Done(err)
+						return err
+					}
+
+					_, err = dhCert.WriteString(d.localConfig["security.sev.session.dh"])
+					if err != nil {
+						op.Done(err)
+						return err
+					}
+
+					dhCertFD = d.addFileDescriptor(&fdFiles, dhCert)
+					defer dhCert.Close()
+
+					dhCertFDOk = true
+				}
+
+				if d.localConfig["security.sev.session.data"] != "" {
+					sessionData, err := os.CreateTemp("", "lxd_sev_session_data_")
+					if err != nil {
+						op.Done(err)
+						return err
+					}
+
+					err = os.Remove(sessionData.Name())
+					if err != nil {
+						op.Done(err)
+						return err
+					}
+
+					_, err = sessionData.WriteString(d.localConfig["security.sev.session.data"])
+					if err != nil {
+						op.Done(err)
+						return err
+					}
+
+					sessionDataFD = d.addFileDescriptor(&fdFiles, sessionData)
+					defer sessionData.Close()
+
+					sessionDataFDOk = true
+				}
+
+				addSEVCmd := func(policy string) {
+					var optSessionFiles string
+					if dhCertFDOk && sessionDataFDOk {
+						optSessionFiles = fmt.Sprintf("dh-cert-file=/proc/fd/%d,session-file=/proc/fd/%d,", dhCertFD, sessionDataFD)
+					}
+
+					qemuCmd = append(qemuCmd, "-machine", "q35,confidential-guest-support=sev0")
+					qemuCmd = append(
+						qemuCmd,
+						"-object",
+						fmt.Sprintf(
+							"sev-guest,id=sev0,cbitpos=%d,reduced-phys-bits=%d,%spolicy=%s",
+							cbitpos,
+							reducedPhysBits,
+							optSessionFiles,
+							policy,
+						),
+					)
+				}
+
+				if shared.IsTrue(d.localConfig["security.sev.policy.es"]) {
+					if slices.Contains(features, "sev-es") {
+						addSEVCmd("0x5")
+					} else {
+						d.logger.Warn("You tried to enable AMD SEV-ES but the host doesn't support it, ignoring")
```

Comment:
Should be `Host system doesn't support AMD SEV-ES` and this should be an error that prevents startup.